lib/fakefs/base.rb

```diff
@@ -78,18 +94,18 @@ def clear!
     end
 
     # present a fresh new fake filesystem to the block
-    def with_fresh(&block)
+    def with_fresh(io_mocks: false, &block)
       clear!
-      with(&block)
+      with(io_mocks: io_mocks, &block)
     end
 
     # present the fake filesystem to the block
-    def with
-      if activated?
+    def with(io_mocks: false)
+      if activated?(desired_io_mocks_state: io_mocks)
```

Comment:
would be nice to be consistent 
```suggestion
      if activated?(io_mocks: io_mocks)
```

Reply:
Done ✅